packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -164,22 +213,21 @@ function serializeCss(result: Result, collectVarFunctions: boolean): string {
         } else if (node && node.type === 'rule' && nodePosition === 'end') {
             currentRuleTokens.push({ type: TokenType.text, value: part });
 
-            // If we are in fakeShadow we dont want to have :host selectors
-            if ((node as any)._isHostNative) {
-                // create an expression for all the tokens (concatenation of strings)
-                // Save it so in the next rule we can apply a ternary
-                tmpHostExpression = generateExpressionFromTokens(currentRuleTokens);
-            } else if ((node as any)._isFakeNative) {
-                const exprToken = generateExpressionFromTokens(currentRuleTokens);
+            // If we are in synthetic shadow we don't want to have :host selectors
+            if ((node as any)._isNativeHost) {
```

Comment:
Could we add a comment here that indicates `postcss-lwc-plugin` will set up the ordering of the nodes so `_isNativeHost` will always appear before a `_isSyntheticHost`?

Reply:
Done!